### ISSUE TITLE
Standard main wrapper for all gRPC plugins and language hosts.

### ISSUE THREAD

**Issue by im7mortal** [pulumi/pulumi#17242]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Regarding  [standard cmd framework](https://github.com/pulumi/pulumi/pull/17129/commits/125b7c5711f8227d9110485e76a61a4dc0b3ae3c#diff-45e854df5c43a84b9e083650d13ead3b154a017851a6dae8c192b2b05808770b) from  #17129.
We have many gRPC servers that share the same logic: creating the gRPC server with standard or provided options, setting live checks, setting tracing handlers, starting the server, printing the port, and performing a proper shutdown. [pulumi-language-java](https://github.com/pulumi/pulumi-java/blob/main/pkg/cmd/pulumi-language-java/main.go), [pulumi-language-dotnet](https://github.com/pulumi/pulumi-dotnet/blob/main/pulumi-language-dotnet/main.go), [pulumi-language-go](https://github.com/pulumi/pulumi/blob/master/sdk/go/pulumi-language-go/main.go), [pulumi-language-python](https://github.com/pulumi/pulumi/blob/master/sdk/python/cmd/pulumi-language-python/main.go), [pulumi-language-nodejs](https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/cmd/pulumi-language-nodejs/main.go).

There are likely more instances where this logic is duplicated.

Additionally, this same approach should be applied for Golang policy and any future gRPC plugins.

Wouldn't it be worth implementing a wrapper in the style of the command framework from this pull request? Of course, with more abstraction.

**Comment by Frassle** [comment]
> Wouldn't it be worth implementing a wrapper in the style of the command framework from this pull request? Of course, with more abstraction.

Yes probably, there's probably enough plugins in the pulumi/pulumi sdk to get a good feel for what's needed here. Pretty much just search for all the places we currently use `rpcutil.ServeWithOptions`.

I've looked into it before and couldn't quickly come up with a good way to express the abstraction given the tools Go gives you, but it might be possible with some more thinking.

**Comment by im7mortal** [comment]
The CMDs which will have the wrapper.



- [X] [pulumi-language-go](https://github.com/pulumi/pulumi/blob/d30f4b5cf6b3523a5f6853a719feec05d2d468b6/sdk/go/pulumi-language-go/main.go#L193)
- [ ] [pulumi-language-python](https://github.com/pulumi/pulumi/blob/d30f4b5cf6b3523a5f6853a719feec05d2d468b6/sdk/python/cmd/pulumi-language-python/main.go#L164)
- [ ] [pulumi-language-java](https://github.com/pulumi/pulumi-java/blob/main/pkg/cmd/pulumi-language-java/main.go)
- [ ] [pulumi-language-nodejs](https://github.com/pulumi/pulumi/blob/d30f4b5cf6b3523a5f6853a719feec05d2d468b6/sdk/nodejs/cmd/pulumi-language-nodejs/main.go#L134)
- [ ] [pulumi-language-dotnet](https://github.com/pulumi/pulumi-dotnet/blob/0436646ae106c1ddc136edcf8cfb9a420031ef94/pulumi-language-dotnet/main.go#L114)
- [ ] [pulumi-converter-constructor-syntax](https://github.com/pulumi/pulumi-converter-constructor-syntax/blob/659af4054fe2454849cf152e5621e3a8d46fba33/cmd/pulumi-converter-constructor-syntax/main.go#L194)
- [ ] [pulumi-converter-terraform](https://github.com/pulumi/pulumi-converter-terraform/blob/aec634efd2ab2601349dbe3ceee34f95313ead65/cmd/pulumi-converter-terraform/main.go#L176)
- [ ] [pulumi-converter-yaml](https://github.com/pulumi/pulumi-yaml/blob/5e605a7c8032647d3282875fbb4f49bb15c33553/cmd/pulumi-converter-yaml/main.go#L108)
- [ ] [pulumi-language-yaml](https://github.com/pulumi/pulumi-yaml/blob/5e605a7c8032647d3282875fbb4f49bb15c33553/cmd/pulumi-language-yaml/main.go#L65)
- [ ] [resource provider](https://github.com/pulumi/pulumi/blob/d30f4b5cf6b3523a5f6853a719feec05d2d468b6/pkg/resource/provider/main.go#L76)
- [ ] [pulumi-analyzer-policy-opa](https://github.com/pulumi/pulumi-policy-opa/blob/df4523cb60d9561a097bc3f7a18119b74b11a8db/cmd/pulumi-analyzer-policy-opa/serve.go#L47)
- [X] pulumi-analyzer-policy-go
